Sweepar.js

eshangonemad opened the pull request:
# Your checklist for this pull request

## Author name
Vasipalli Eshan Aditya
**Author:**
eshangonemad
## About your game
Sweepar is a fun and suspenseful game that makes players follow their instincts
**What is your game about?**
Sweepar is a hybrid minesweeper-inspired game that leaves the decision to its users, think of it like minesweeper without help. Sweepar uses a tiny cursor that moves around and lets users take control of the board

**How do you play your game?**
WASD to move the cursor around the grid, J to open squares, K to flag squares and I to reset the game

## Code
Check off the items that are true. <!-- Put a `x` in the  `[ ]`. Example: `[x] The game was made...` -->
- [x ] The game was made using the [Sprig editor](https://sprig.hackclub.com/editor).
- [x ] The game is placed in the in the [`/games` directory](https://github.com/hackclub/sprig/tree/main/games).
- [ x] The code is significantly different from all other games in the [Sprig gallery](https://sprig.hackclub.com/gallery) (except for games labeled "demo").
- [ x] The game runs without errors. 
- [ x] The name of the file/game contains only alphanumeric characters, `-`s, or `_`s.
- [ x] The game name is not the same as the others from [gallery](https://sprig.hackclub.com/gallery)

## Image (If an image is used)

- [x] The image is in the [`/games/img` directory](https://github.com/hackclub/sprig/tree/main/games/img).
- [ ] The name of the image matches the name of your file. <!-- Example: `sokoban_plus.js` -> `sokoban_plus.png`. -->

> Thanks for your PR!

LucasHT22 reviewed (changes requested):
Thanks for the PR! Where did you hear about Sprig?

eshangonemad commented:
I learnt about sprig on the hackclub youtube


On Sat, Mar 9, 2024 at 10:16 AM Lucas Honda ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Thanks for the PR! Where did you hear about Sprig?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1926072352>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMQ5WHPTQNOXCEFL4HTYXJWGJAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRWGA3TEMZVGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

LucasHT22 commented:
> I learnt about sprig on the hackclub youtube
> […](#)
> On Sat, Mar 9, 2024 at 10:16 AM Lucas Honda ***@***.***> wrote: ***@***.**** requested changes on this pull request. Thanks for the PR! Where did you hear about Sprig? — Reply to this email directly, view it on GitHub <[#1505 (review)](https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1926072352)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AT4KHMQ5WHPTQNOXCEFL4HTYXJWGJAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRWGA3TEMZVGI> . You are receiving this because you were mentioned.Message ID: ***@***.***>

oh nice! the youtube shorts?

eshangonemad commented:
Yep

On Sun, Mar 10, 2024 at 8:43 AM Lucas Honda ***@***.***>
wrote:

> I learnt about sprig on the hackclub youtube
> … <#m_-7754240004917809218_>
> On Sat, Mar 9, 2024 at 10:16 AM Lucas Honda *@*.*> wrote: @.** requested
> changes on this pull request. Thanks for the PR! Where did you hear about
> Sprig? — Reply to this email directly, view it on GitHub <#1505 (review)
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1926072352>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AT4KHMQ5WHPTQNOXCEFL4HTYXJWGJAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRWGA3TEMZVGI
> . You are receiving this because you were mentioned.Message ID: *@*.***>
>
> oh nice! the youtube shorts?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-1987026150>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMSLFKNEEXT2WFJYYLDYXOUCPAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBXGAZDMMJVGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

LucasHT22 reviewed (changes requested):
Nice! What about add instructions to the header?

eshangonemad commented:
i already added that right?

On Mon, Mar 11, 2024 at 3:28 AM Lucas Honda ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Nice! What about add instructions to the header?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1926684445>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMUY3RAHKDXX2QTG3C3YXSX5XAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRWGY4DINBUGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

LucasHT22 commented:
> i already added that right?
> […](#)
> On Mon, Mar 11, 2024 at 3:28 AM Lucas Honda ***@***.***> wrote: ***@***.**** requested changes on this pull request. Nice! What about add instructions to the header? — Reply to this email directly, view it on GitHub <[#1505 (review)](https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1926684445)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AT4KHMUY3RAHKDXX2QTG3C3YXSX5XAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRWGY4DINBUGU> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Go to `Files changed` > `...` > `edit` :)

eshangonemad commented:
i added it


On Tue, Mar 12, 2024 at 8:08 AM Lucas Honda ***@***.***>
wrote:

> i already added that right?
> … <#m_-7604705876421607008_>
> On Mon, Mar 11, 2024 at 3:28 AM Lucas Honda *@*.*> wrote: @.** requested
> changes on this pull request. Nice! What about add instructions to the
> header? — Reply to this email directly, view it on GitHub <#1505 (review)
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1926684445>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AT4KHMUY3RAHKDXX2QTG3C3YXSX5XAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRWGY4DINBUGU
> . You are receiving this because you were mentioned.Message ID: *@*.***>
>
> Go to Files changed > ... > edit :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-1989672699>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMXKV2GWP2QQOPIW2QLYXZBQTAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBZGY3TENRZHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

LucasHT22 reviewed (changes requested):
Right, Out of curiosity, did you use any game for inspiration?

eshangonemad commented:
i took inspiration from the minesweeper game i love

On Wed, Mar 13, 2024 at 1:33 AM Lucas Honda ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Right, Out of curiosity, did you use any game for inspiration?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1931801758>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMRBIOJSI56EWSC5NGTYX436VAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZRHAYDCNZVHA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

LucasHT22 commented:
> i took inspiration from the minesweeper game i love
> […](#)
> On Wed, Mar 13, 2024 at 1:33 AM Lucas Honda ***@***.***> wrote: ***@***.**** requested changes on this pull request. Right, Out of curiosity, did you use any game for inspiration? — Reply to this email directly, view it on GitHub <[#1505 (review)](https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1931801758)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AT4KHMRBIOJSI56EWSC5NGTYX436VAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZRHAYDCNZVHA> . You are receiving this because you commented.Message ID: ***@***.***>

Right:) but did you use any game from our gallery as inspiration?

eshangonemad commented:
No

On Thu, Mar 14, 2024 at 1:24 AM Lucas Honda ***@***.***>
wrote:

> i took inspiration from the minesweeper game i love
> … <#m_968241358896364154_>
> On Wed, Mar 13, 2024 at 1:33 AM Lucas Honda *@*.*> wrote: @.** requested
> changes on this pull request. Right, Out of curiosity, did you use any game
> for inspiration? — Reply to this email directly, view it on GitHub <#1505
> (review)
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1931801758>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AT4KHMRBIOJSI56EWSC5NGTYX436VAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZRHAYDCNZVHA
> . You are receiving this because you commented.Message ID: *@*.***>
>
> Right:) but did you use any game from our gallery as inspiration?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-1995076027>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMWDWWAEKXRD36Y5L2LYYCDTHAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOJVGA3TMMBSG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

LucasHT22 commented:
Right! did you beat your game?

eshangonemad commented:
Yes, I did. it was challenging tbh

On Fri, Mar 15, 2024 at 8:06 AM Lucas Honda ***@***.***>
wrote:

> Right! did you beat your game?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-1998673278>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMUOLALVDVNRBYRX3WTYYI3SDAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOJYGY3TGMRXHA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

LucasHT22 commented:
> Yes, I did. it was challenging tbh

Nice! Do you have a screenshot?

eshangonemad commented:
unfortunately, i do not have any screenshots

On Sat, Mar 16, 2024 at 2:07 AM Lucas Honda ***@***.***>
wrote:

> Yes, I did. it was challenging tbh
>
> Nice! Do you have a screenshot?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-2000185359>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMXHH5QERHONU75AF33YYM2HNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DKMZVHE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

LucasHT22 commented:
> unfortunately, i do not have any screenshots
> […](#)
> On Sat, Mar 16, 2024 at 2:07 AM Lucas Honda ***@***.***> wrote: Yes, I did. it was challenging tbh Nice! Do you have a screenshot? — Reply to this email directly, view it on GitHub <[#1505 (comment)](https://github.com/hackclub/sprig/pull/1505#issuecomment-2000185359)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AT4KHMXHH5QERHONU75AF33YYM2HNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DKMZVHE> . You are receiving this because you commented.Message ID: ***@***.***>

Hmmm could you try again?

eshangonemad commented:
[image: image.png]

On Sun, Mar 17, 2024 at 9:00 AM Lucas Honda ***@***.***>
wrote:

> unfortunately, i do not have any screenshots
> … <#m_-6862090769286791921_>
> On Sat, Mar 16, 2024 at 2:07 AM Lucas Honda *@*.*> wrote: Yes, I did. it
> was challenging tbh Nice! Do you have a screenshot? — Reply to this email
> directly, view it on GitHub <#1505 (comment)
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-2000185359>>, or
> unsubscribe
> https://github.com/notifications/unsubscribe-auth/AT4KHMXHH5QERHONU75AF33YYM2HNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DKMZVHE
> <https://github.com/notifications/unsubscribe-auth/AT4KHMXHH5QERHONU75AF33YYM2HNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DKMZVHE>
> . You are receiving this because you commented.Message ID: @.*>
>
> Hmmm could you try again?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-2002215469>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMXVZ5K4B5ZG73AXIEDYYTTMNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBSGIYTKNBWHE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

LucasHT22 commented:
> [image: image.png] On Sun, Mar 17, 2024 at 9:00 AM Lucas Honda ***@***.***> wrote:
> […](#)
> unfortunately, i do not have any screenshots … <#m_-6862090769286791921_> On Sat, Mar 16, 2024 at 2:07 AM Lucas Honda *@*.*> wrote: Yes, I did. it was challenging tbh Nice! Do you have a screenshot? — Reply to this email directly, view it on GitHub <#1505 (comment) <[#1505 (comment)](https://github.com/hackclub/sprig/pull/1505#issuecomment-2000185359)>>, or unsubscribe https://github.com/notifications/unsubscribe-auth/AT4KHMXHH5QERHONU75AF33YYM2HNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DKMZVHE <https://github.com/notifications/unsubscribe-auth/AT4KHMXHH5QERHONU75AF33YYM2HNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DKMZVHE> . You are receiving this because you commented.Message ID: @.*> Hmmm could you try again? — Reply to this email directly, view it on GitHub <[#1505 (comment)](https://github.com/hackclub/sprig/pull/1505#issuecomment-2002215469)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AT4KHMXVZ5K4B5ZG73AXIEDYYTTMNAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBSGIYTKNBWHE> . You are receiving this because you commented.Message ID: ***@***.***>

The image didn't load for me, could you re-send?

eshangonemad commented:
![Screenshot 2024-03-17 131643](https://github.com/hackclub/sprig/assets/83403698/b6a589c6-d8f1-4744-b198-38d57a9ff9a6)

eshangonemad commented:
https://youtu.be/0qjz6Mtu6H0

LucasHT22 commented:
Thanks! What about adding a clearText() in restart button?

eshangonemad commented:
Done I added it

On Wed, Mar 20, 2024, 2:02 AM Lucas Honda ***@***.***> wrote:

> Thanks! What about adding a clearText() in restart button?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#issuecomment-2007817924>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMRXUROF7U7CTY3TQGLYZB4UHAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBXHAYTOOJSGQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

LucasHT22 reviewed (approved):
Ok! Congrats:)
If you're a teen, you can get your sprig here! https://sprig-order.hackclub.dev/

Hack Clubbers would love to see this! Just post a link and 1-2 sentences talking about your game to our [#ship](https://hackclub.slack.com/archives/C0M8PUPU6) [channel](https://hackclub.com/slack/):)

eshangonemad commented:
thank you :)

On Thu, Mar 21, 2024 at 2:08 AM Lucas Honda ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Ok! Congrats:)
> If you're a teen, you can get your sprig here!
> https://sprig-order.hackclub.dev/
>
> Hack Clubbers would love to see this! Just post a link and 1-2 sentences
> talking about your game to our #ship
> <https://hackclub.slack.com/archives/C0M8PUPU6> channel
> <https://hackclub.com/slack/>:)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/sprig/pull/1505#pullrequestreview-1949692239>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4KHMRY6FOBTB6MEXVNXZLYZHGCZAVCNFSM6AAAAABENWRETGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNBZGY4TEMRTHE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
